File: prisma/seed.ts
import { PrismaClient, Role } from "@prisma/client";
import bcrypt from "bcryptjs";

const prisma = new PrismaClient();

async function main() {
  const adminPassword = await bcrypt.hash("admin1234", 10);
  const memberPassword = await bcrypt.hash("1", 10);

  await prisma.user.upsert({
    where: { email: "admin@golf.com" },
    update: { dong: "0", ho: "admin" },
    create: {
      email: "admin@golf.com",
      passwordHash: adminPassword,
      name: "관리자",
      dong: "0",
      ho: "admin",
      role: Role.ADMIN,
      isActive: true,
    },
  });

  const memberEmail = "101-1001@member.golf";
  await prisma.user.upsert({
    where: { email: memberEmail },
    update: {
      dong: "101",
      ho: "1001",
      passwordHash: memberPassword,
      name: "홍길동",
    },
    create: {
      email: memberEmail,
      passwordHash: memberPassword,
      name: "홍길동",
      dong: "101",
      ho: "1001",
      role: Role.USER,
      isActive: true,
    },
  });

  console.log("Seed completed:");
  console.log("  Admin: admin@golf.com / admin1234");
  console.log("  Member: 101동 1001호 / password: 1");
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

File: backend/seed.py
"""Seed initial admin and test member accounts."""

import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from app.database import SessionLocal
from app.models import Role, User
from app.services.formatting import hash_password


def main() -> None:
    db = SessionLocal()
    try:
        admin = db.query(User).filter(User.email == "admin@golf.com").first()
        if admin:
            admin.dong = "0"
            admin.ho = "admin"
        else:
            admin = User(
                email="admin@golf.com",
                passwordHash=hash_password("admin1234"),
                name="관리자",
                dong="0",
                ho="admin",
                role=Role.ADMIN,
                isActive=True,
            )
            db.add(admin)

        member_email = "101-1001@member.golf"
        member = db.query(User).filter(User.email == member_email).first()
        if member:
            member.dong = "101"
            member.ho = "1001"
            member.phone = "01012345678"
            member.passwordHash = hash_password("1")
            member.name = "홍길동"
        else:
            member = User(
                email=member_email,
                passwordHash=hash_password("1"),
                name="홍길동",
                phone="01012345678",
                dong="101",
                ho="1001",
                role=Role.USER,
                isActive=True,
            )
            db.add(member)

        db.commit()
        print("Seed completed:")
        print("  Admin: admin@golf.com / admin1234")
        print("  Member: 101동 1001호 / password: 1")
    finally:
        db.close()


if __name__ == "__main__":
    main()
